tool/tbot/main.go

```diff
@@ -271,20 +236,72 @@ func watchCARotations(watcher types.Watcher) {
 	}
 }
 
+func getIdentityFromToken(cfg *config.BotConfig) (*identity.Identity, error) {
+	if cfg.Onboarding == nil {
+		return nil, trace.BadParameter("onboarding config required via CLI or YAML")
+	}
+	if cfg.Onboarding.Token == "" {
+		return nil, trace.BadParameter("unable to start: no token present")
+	}
+	addr, err := utils.ParseAddr(cfg.AuthServer)
+	if err != nil {
+		return nil, trace.WrapWithMessage(err, "invalid auth server address %+v", cfg.AuthServer)
+	}
+
+	tlsPrivateKey, sshPublicKey, tlsPublicKey, err := generateKeys()
+	if err != nil {
+		return nil, trace.WrapWithMessage(err, "unable to generate new keypairs")
+	}
+
+	log.Info("attempting to generate new identity from token")
+	params := auth.RegisterParams{
+		Token: cfg.Onboarding.Token,
+		ID: auth.IdentityID{
+			Role: types.RoleBot,
+		},
+		Servers:            []utils.NetAddr{*addr},
+		PublicTLSKey:       tlsPublicKey,
+		PublicSSHKey:       sshPublicKey,
+		CAPins:             cfg.Onboarding.CAPins,
+		CAPath:             cfg.Onboarding.CAPath,
+		GetHostCredentials: client.HostCredentials,
+		JoinMethod:         cfg.Onboarding.JoinMethod,
+	}
+	certs, err := auth.Register(params)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	ident, err := identity.ReadIdentityFromKeyPair(tlsPrivateKey, sshPublicKey, certs)
+	return ident, trace.Wrap(err)
+}
+
 func renewIdentityViaAuth(
 	ctx context.Context,
 	client auth.ClientI,
 	currentIdentity *identity.Identity,
-	certTTL time.Duration,
+	cfg *config.BotConfig,
 ) (*identity.Identity, error) {
 	// TODO: enforce expiration > renewal period (by what margin?)
 
+	// If using the EC2 or IAM join method, repeatedly fetch new certs rather
+	// than renewing.
```

Comment:
"Repeatedly" here describes the overall logic, right? Like, for other join methods we repeatedly renew certs, but for EC2 and IAM we repeatedly fetch new certs?

Reply:
Yes that's right, I've updated the comment to try to make it a bit more clear. Thanks for reviewing!